src/txn/txn_log.c

```diff
@@ -78,31 +77,47 @@ __txn_op_log(WT_SESSION_IMPL *session,
 	value.size = upd->size;
 
 	/*
-	 * Log the operation. It must be a row- or column-store insert, remove
-	 * or update, all of which require log records. We shouldn't ever log
-	 * reserve operations.
+	 * Log the row- or column-store insert, modify, remove or update. Our
+	 * caller doesn't log reserve operations, we shouldn't see them here.
 	 */
-	WT_ASSERT(session, upd->type != WT_UPDATE_RESERVED);
 	if (cbt->btree->type == BTREE_ROW) {
 #ifdef HAVE_DIAGNOSTIC
 		__txn_op_log_row_key_check(session, cbt);
 #endif
-		if (upd->type == WT_UPDATE_DELETED)
+		switch (upd->type) {
+		case WT_UPDATE_STANDARD:
```

Comment:
> Or did you put STANDARD first because it is expected to be the most common?

It's hold-over from when we didn't pass a `WT_UPDATE` type into the row-/column-store modification routines, so 0 was the "default" type.

That's no longer the case, and we could sort all of the places we use `upd->type`, but I think it makes the code a little less readable, having the common case first helps a little.

I doubt there's any performance reason for having the `WT_UPDATE_STANDARD` first, since the type value is 0, 1, 2, or 3, it's going to be compiled as a jump table, I think.

tl;dr: it's a holdover from an old implementation, let me know if you want it changed.